src/app/vendors/reducer.ts

```diff
@@ -38,7 +40,37 @@ export const vendors: Reducer<VendorsState, VendorsAction | AccountsAction> = (
           ...state.vendorsByCharacter,
           [characterId]: {
             vendorsResponse: vendorsResponse,
-            lastLoaded: new Date(),
+            lastLoaded: Date.now(),
+            error: undefined,
+          },
+        },
+      };
+    }
+
+    // augments the storedoverall all-vendors response,
+    // by inserting the item components from a single-vendor api response
+    case getType(actions.loadedSingle): {
+      const { characterId, vendorResponse, vendorHash } = action.payload;
+      const { vendorsByCharacter } = state;
+      const oldVendorsResponse = vendorsByCharacter[characterId]?.vendorsResponse;
+
+      // nothing about state needs changing if we didn't get back components
+      // (maybe sockets/etc are disabled at bnet right now?)
+      if (_.isEmpty(vendorResponse.itemComponents) || !oldVendorsResponse) {
+        return state;
+      }
+
+      const newItemComponents = vendorResponse.itemComponents; // big combined set, keyed by vendor hash
+      const existingItemComponents = oldVendorsResponse?.itemComponents; // set for a single vendor
+      const itemComponents = { ...existingItemComponents, [vendorHash]: newItemComponents }; // big set again
+
+      return {
+        ...state,
+        vendorsByCharacter: {
+          ...vendorsByCharacter,
+          [characterId]: {
+            vendorsResponse: { ...oldVendorsResponse, itemComponents },
+            lastLoaded: Date.now(),
             error: undefined,
           },
         },
```

Comment:
This is all perfectly reasonable, though the `immer` version might read better:
```suggestion
      return produce(state, (draft) => {
        draft.vendorsByCharacter[characterId].vendorsResponse.itemComponents[vendorHash] = vendorResponse.itemComponents;
        draft.vendorsByCharacter[characterId].lastLoaded = Date.now();
      }),
```